src/biomapper2/core/annotators/metabolomics_workbench.py

```diff
@@ -0,0 +1,138 @@
+"""Metabolomics Workbench RefMet API annotator for metabolite entities."""
+
+import logging
+from collections import defaultdict
+from typing import Any
+from urllib.parse import quote
+
+import pandas as pd
+import requests
+
+from ...utils import AssignedIDsDict
+from .base import BaseAnnotator
+
+
+class MetabolomicsWorkbenchAnnotator(BaseAnnotator):
+    """Annotator that queries the Metabolomics Workbench RefMet API.
+
+    Retrieves vocabulary IDs (PubChem, InChIKey, SMILES, RefMet) for metabolite entities.
+
+    API Endpoint: GET https://www.metabolomicsworkbench.org/rest/refmet/name/{metabolite_name}/all/
+    """
+
+    slug = "metabolomics-workbench"
+    BASE_URL = "https://www.metabolomicsworkbench.org/rest/refmet/name"
+
+    # Mapping from API response fields to normalizer vocabulary names
+    FIELD_TO_VOCAB = {
+        "pubchem_cid": "pubchem.compound",
+        "inchi_key": "inchikey",
+        "smiles": "smiles",
+        "refmet_id": "rm",
+    }
+
+    def get_annotations(self, entity: dict | pd.Series, name_field: str, cache: dict | None = None) -> AssignedIDsDict:
+        """Get annotations for a single entity.
+
+        Args:
+            entity: Entity to annotate (dict or DataFrame row)
+            name_field: Name of the field containing the entity name
+            cache: Optional pre-fetched results from bulk API call
+
+        Returns:
+            Dict with annotation results
+        """
+        # Extract the entity name
+        name = entity.get(name_field)
+
+        if not name:
+            # No name provided, cannot annotate
+            return {}
+
+        # Use cache if available, otherwise fetch from API
+        if cache is not None:
+            api_data = cache.get(name)
+        else:
+            api_data = self._fetch_refmet_data(name)
+
+        if not api_data:
+            # No data returned from API
+            return {self.slug: {}}
+
+        # Build the annotations structure
+        annotations: dict[str, dict[str, dict[str, Any]]] = defaultdict(lambda: defaultdict(dict))
+
+        for api_field, vocab in self.FIELD_TO_VOCAB.items():
+            value = api_data.get(api_field)
+            if value:
+                # Clean up the value (e.g., remove "RM" prefix from refmet_id)
+                local_id = self._clean_value(api_field, value)
```

Comment:
I'd suggest removing this local ID cleaning step - again this is something that is centralized in the `Normalizer` - it already handles refmet local IDs that include the 'RM' prefix (better not to duplicate that work here)